[docs] Update app-credentials.mdx (#24631)
expo/expo: docs/pages/app-signing/app-credentials.mdx

### Re-signing new credentials

> You can use `eas build:resign` to resign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally &mdash; for example, if you want to add a new test device to an existing build, you can use this command to update the provisioning profile to include the device without rebuilding the entire app from scratch.

Reviewer: ```suggestion
You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally &mdash; for example, if you want to add a new test device to an existing build, you can use this command to update the provisioning profile to include the device without rebuilding the entire app from scratch.
```

@albert-schilling, the terminology Apple uses in their official documentation for this process is Code signing: https://developer.apple.com/documentation/security/code_signing_services. Sticking to that terminology to describe what the command does, I'm reverting this suggestion.

After selecting the build, follow the steps to log in to your Apple Developer account. When prompted **Show devices and ask me again**, you can select a new provisioning profile.

> Select a new device, and the command will run the EAS Build again. Note that the build triggered this time reuses the application artifact from the selected build and resigns it with the new provisioning profile. Once the process of resigning is complete, you can use this new build link to install the **.ipa** on the iOS device that was added to the provisioning profile during the process.

Reviewer: ```suggestion
Select a new device, and the command will run the EAS Build again. Note that the build triggered this time reuses the application artifact from the selected build and codesigns it with the new provisioning profile. Once this process is complete, you can use this new build link to install the **.ipa** on the iOS device added to the provisioning profile.
```

I think the confusing part here is the last line but again, in the second line, the process of code signing an ipa is being described so let's stick to the terminology.
